include/zephyr/llext/buf_loader.h

```diff
@@ -56,11 +49,72 @@ void *llext_buf_peek(struct llext_loader *ldr, size_t pos);
 				.seek = llext_buf_seek,                                            \
 				.peek = llext_buf_peek,                                            \
 				.finalize = NULL,                                                  \
+				.storage = _storage,                                               \
 			},                                                                         \
 		.buf = (_buf),                                                                     \
 		.len = (_buf_len),                                                                 \
 		.pos = 0,                                                                          \
 	}
+/** @endcond */
+
+/**
+ * @brief Initializer for an llext_buf_loader structure
+ *
+ * The storage type for the provided buffer depends on the value of the
+ * @kconfig{CONFIG_LLEXT_STORAGE_WRITABLE} option: if it is defined, the
+ * buffer is assumed to be writable; otherwise it is assumed to be persistent.
+ *
+ * Consider using one of the alternative macros instead.
+ *
+ * @see LLEXT_TEMPORARY_BUF_LOADER
+ * @see LLEXT_PERSISTENT_BUF_LOADER
+ * @see LLEXT_WRITABLE_BUF_LOADER
+ *
+ * @param _buf Buffer containing the ELF binary
+ * @param _buf_len Buffer length in bytes
+ */
+#define LLEXT_BUF_LOADER(_buf, _buf_len) \
```

Comment:
I'm in favor of deprecating this given the variants are now available at construction time